### DISCUSSION TITLE
openai.ChatCompletion deprecated!

### DISCUSSION THREAD

**Discussion by Privat3m** [knrd1/chatgpt#40] [Q&A]
Hi
As soon as a query is sent to my bot, it gets disconnected with this error;
You tried to access openai.ChatCompletion, but this is no longer supported in openai>=1.0.0

cant downgrade; 
'Found existing installation: openai 1.12.0
ERROR: Cannot uninstall openai 1.12.0, RECORD file not found. Hint: The package was installed by debian.'

**Comment by knrd1** [comment]
Yeah, the code needs to be rewritten to support newer openai versions and I'm planning to do that soon.
For now, downgrade to 0.28, I have it on my Debian 12:

root@aws:~# pip3 list|grep openai
openai              0.28.0

Try this:

pip3 uninstall openai --break-system-packages
pip3 install openai==0.28 --break-system-packages
